### PR TITLE
forward basic auth to OpenID connect token authentication endpoint

### PR DESCRIPTION
**Pull request by butonic**
fixes https://github.com/owncloud/ocis/issues/2094

### CODE REVIEW

File: proxy/pkg/middleware/basic_auth.go [hunk 1]
```diff
@@ -89,6 +90,12 @@ func (m basicAuth) isPublicLink(req *http.Request) bool {
 	return ok && login == "public" && strings.HasPrefix(req.URL.Path, publicFilesEndpoint)
 }
 
+// The token auth endpoint uses basic auth for clients, see https://openid.net/specs/openid-connect-basic-1_0.html#TokenRequest
+// > The Client MUST authenticate to the Token Endpoint using the HTTP Basic method, as described in 2.3.1 of OAuth 2.0.
+func (m basicAuth) isOIDCTokenAuth(req *http.Request) bool {
+	return req.URL.Path == "/konnect/v1/token"
```
Comment (C0rby) [inline proxy/pkg/middleware/basic_auth.go:96]: This is now hard-coded for konnectd. What about other IDPs?

Comment (butonic) [reply proxy/pkg/middleware/basic_auth.go:96]: Other IdPs won't be hidden behind our proxy.

Comment (C0rby) [reply proxy/pkg/middleware/basic_auth.go:96]: I see. Then it should be ok, I guess.

**Review by C0rby — APPROVED** [review]
Regarding the failing tests. Maybe rebasing will help?

**Comment by butonic** [comment]
needs rebase

**Pull request merged**